packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { toPng } from 'html-to-image';
+import {
+    Button,
+    Clipboard,
+    DesktopWrapper,
+    Icon,
+    MobileWrapper,
+    Modal,
+    Text,
+    useCopyToClipboard,
+} from '@deriv/components';
+import { websiteUrl } from '@deriv/shared';
+import { observer } from '@deriv/stores';
+import { Localize, localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import MyProfileSeparatorContainer from 'Components/my-profile/my-profile-separator-container';
+import { buy_sell } from 'Constants/buy-sell';
+import { ad_type } from 'Constants/floating-rate';
+import { TAdvert } from 'Types';
+import ShareMyAdsCard from './share-my-ads-card';
+import ShareMyAdsSocials from './share-my-ads-socials';
+
+const ShareMyAdsModal = ({ advert }: TAdvert) => {
+    const [is_copied, copyToClipboard, setIsCopied] = useCopyToClipboard();
+    const { account_currency, advertiser_details, id, local_currency, rate_display, rate_type, type } = advert;
+    const { id: advertiser_id } = advertiser_details;
+    const { hideModal, is_modal_open } = useModalManagerContext();
+
+    const divRef = React.useRef(null);
+    const advert_url = `${websiteUrl()}cashier/p2p/advertiser?id=${advertiser_id}&advert_id=${id}`;
+    const is_buy_ad = type === buy_sell.BUY;
+    const custom_message = localize(
+        "Hi! I'd like to exchange {{first_currency}} for {{second_currency}} at {{rate_display}}{{rate_type}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+        {
+            first_currency: is_buy_ad ? local_currency : account_currency,
+            second_currency: is_buy_ad ? account_currency : local_currency,
+            rate_display,
+            rate_type: rate_type === ad_type.FLOAT ? '%' : ` ${local_currency}`,
+            advert_url,
+        }
+    );
```

Comment:
will this work?
```
  <Localize
                    i18n_default_text="Hi! I'd like to exchange {{first_currency}} for {{second_currency}} at {{rate_display}}{{rate_type}} on Deriv P2P.<0/>If you're interested, check out my ad 👉<0/>{{- advert_url}}<0/>Thanks!"
                    components={[<br key={0} />]}
                    values={{
                        first_currency: is_buy_ad ? local_currency : account_currency,
                        second_currency: is_buy_ad ? account_currency : local_currency,
                        rate_display,
                        rate_type: rate_type === ad_type.FLOAT ? '%' : ` ${local_currency}`,
                        advert_url,
                    }}
                />
```

Reply:
I think so, let me change this

Reply:
ok just checked, i think its better to use `localize` in this case because when I use `Localize` it sends the message as `[object Object]`... probably due to it being of Element type and localize being string...